homeassistant/components/sum/sensor.py

```diff
@@ -0,0 +1,214 @@
+
+    async_add_entities(
+        [
+            SumSensor(
+                entity_ids,
+                config_entry.title,
+                round_digits,
+                config_entry.entry_id,
+            )
+        ]
+    )
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the Sum sensor."""
+    if discovery_info is None:
+        return
+    entity_ids: list[str] = discovery_info[CONF_ENTITY_IDS]
+    name: str = discovery_info[CONF_NAME]
+    round_digits: float = discovery_info[CONF_ROUND_DIGITS]
+    unique_id: str | None = discovery_info.get(CONF_UNIQUE_ID)
+
+    await async_setup_reload_service(hass, DOMAIN, PLATFORMS)
+
+    async_add_entities([SumSensor(entity_ids, name, round_digits, unique_id)])
+
+
+def calc_sum(sensor_values: list[tuple[str, Any]], round_digits: int) -> float | None:
+    """Calculate sum value, skipping unknown/unavailable state."""
+    result = []
+    for _, sensor_value in sensor_values:
+        if sensor_value in [STATE_UNKNOWN, STATE_UNAVAILABLE]:
+            continue
+        result.append(sensor_value)
+
+    value: float = round(sum(result), round_digits)
+    return value
+
+
+class SumSensor(SensorEntity):
+    """Representation of a sum sensor."""
+
+    _attr_icon = ICON
+    _attr_should_poll = False
+    _attr_state_class = SensorStateClass.MEASUREMENT
+
+    def __init__(
+        self,
+        entity_ids: list[str],
+        name: str,
+        round_digits: float,
+        unique_id: str | None,
+    ) -> None:
+        """Initialize the sum sensor."""
+        self._attr_unique_id = unique_id
+        self._entity_ids = entity_ids
+        self._round_digits = int(round_digits)
+
+        self._attr_name = name
+
+        self.sum: float | None = None
+        self.valid_entities: list[str] | None = None
+        self._unit_of_measurement = None
+        self.states: dict[str, Any] = {}
+
+    async def async_added_to_hass(self) -> None:
+        """Handle added to Hass."""
+        self.async_on_remove(
+            async_track_state_change_event(
+                self.hass, self._entity_ids, self._async_sum_sensor_state_listener
+            )
+        )
+
+        # Replay current state of source entities
+        for entity_id in self._entity_ids:
+            state = self.hass.states.get(entity_id)
+            state_event = Event("", {"entity_id": entity_id, "new_state": state})
+            self._async_sum_sensor_state_listener(state_event, update_state=False)
+
+        self._calc_values()
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the state of the sensor."""
+        return self.sum
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return extra state attributes."""
+        return {
+            ATTR_ENTITIES: self._entity_ids,
```

Comment:
Make sure we add the recorder platform to this integration to ensure this attribute is excluded from the recorder (as it is configuration).